src/config/backends/extra-uEnv.ts

```diff
@@ -0,0 +1,261 @@
+			(typeof metaRelease === 'undefined' ||
+				semver.gte(metaRelease, EXTRA_SUPPORT.jetson))
+		);
+	}
+
+	public async getBootConfig(): Promise<ConfigOptions> {
+		// Get config contents at bootConfigPath
+		const confContents = await ExtraUEnvConfigBackend.readBootConfigPath();
+
+		// Parse ConfigOptions from bootConfigPath contents
+		const parsedConfigFile = ExtraUEnvConfigBackend.parseOptions(confContents);
+
+		// Filter out unsupported values
+		return _.pickBy(parsedConfigFile, (_value, key) =>
+			this.isSupportedConfig(key),
+		);
+	}
+
+	public async setBootConfig(opts: ConfigOptions): Promise<void> {
+		// Filter out unsupported options
+		const supportedOptions = _.pickBy(opts, (value, key) => {
+			if (!this.isSupportedConfig(key)) {
+				log.warn(`Not setting unsupported value: { ${key}: ${value} }`);
+				return false;
+			}
+			return true;
+		});
+
+		// Write new extra_uEnv configuration
+		return await remountAndWriteAtomic(
+			ExtraUEnvConfigBackend.bootConfigPath,
+			ExtraUEnvConfigBackend.configToString(supportedOptions),
+		);
+	}
+
+	public isSupportedConfig(config: string): boolean {
+		return config in ExtraUEnvConfigBackend.supportedConfigs;
+	}
+
+	public isBootConfigVar(envVar: string): boolean {
+		return envVar.startsWith(ExtraUEnvConfigBackend.bootConfigVarPrefix);
+	}
+
+	public processConfigVarName(envVar: string): string | null {
+		const name = envVar.replace(
+			ExtraUEnvConfigBackend.bootConfigVarRegex,
+			'$1',
+		);
+		if (name === envVar) {
+			return null;
+		}
+		return name;
+	}
+
+	public processConfigVarValue(_key: string, value: string): string {
+		return value;
+	}
+
+	public createConfigVarName(configName: string): string | null {
+		if (configName === '') {
+			return null;
+		}
+		return `${ExtraUEnvConfigBackend.bootConfigVarPrefix}${configName}`;
+	}
+
+	private static parseOptions(configFile: string): ConfigOptions {
+		// Split by line and filter any comments and empty lines
+		const lines = configFile.split(LINE_REGEX);
+		// Reduce lines to ConfigOptions
+		return lines.reduce((options: ConfigOptions, line: string) => {
+			const optionValues = line.match(OPTION_REGEX);
+			if (optionValues == null) {
+				log.warn(`Could not read extra_uEnv entry: ${line}`);
+				return options;
+			}
+			// Merge new option with existing options
+			return {
+				...ExtraUEnvConfigBackend.parseOption(optionValues),
+				...options,
+			};
+		}, {});
+	}
+
+	private static parseOption(optionArray: string[]): ConfigOptions {
+		const [, KEY, VALUE] = optionArray;
+		// Check if this key's value is a collection
+		if (ExtraUEnvConfigBackend.entries[KEY as EntryKey]?.collection) {
+			// Return split collection of options
+			return ExtraUEnvConfigBackend.parseOptionCollection(VALUE);
+		}
+		// Find the option that belongs to this entry
+		const optionKey = _.findKey(
+			ExtraUEnvConfigBackend.supportedConfigs,
+			(_v, k) => ExtraUEnvConfigBackend.supportedConfigs[k].key === KEY,
```

Comment:
```suggestion
			(config) => config.key === KEY,
```

Reply:
ahh... Why did I write that... thank you for catching that.